### ISSUE TITLE
Broadcasting with IdOffsetRange seems wrong

### ISSUE THREAD

**Issue by yha** [JuliaArrays/OffsetArrays.jl#104]
This seems fine, although the error message is a bit odd:
```
oa = OffsetArray(1:4,1)
axes(oa .+ 1)  # (2:5)
oa .+ (1:4)   # 
DimensionMismatch("arrays could not be broadcast to a common size; got a dimension with lengths 4 and 4")
```
But:
```
oa_axes = axes(oa)[1]  # 2:5
axes(oa_axes)  # (2:5)
axes(oa_axes .+ 1)  # (Base.OneTo(4))
oa_axes .+ (1:4)  #  3:2:9
```
I would expect `oa_axes` to behaves like `oa` in broadcasting.

**Comment by johnnychen94** [comment]
`+-*/` of two `IdentityUnitRange` isn't an `IdentityUnitRange` anymore, and that's the same fact for `IdOffsetRange`.

```julia
julia> Base.IdentityUnitRange(2:5) .+ (1:4)
3:2:9
```

But `oa_axes .+ 1` should preserve the offset. #105 fixes this.

**Comment by timholy** [comment]
See https://github.com/JuliaLang/julia/pull/30950. That was starting to go a in a scary breaking direction, that effort needs to be restarted with a more careful design.

**Comment by yha** [comment]
Closing since this is fixed by #114.
